apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx

```diff
@@ -0,0 +1,178 @@
+    const IsEmergencyMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isEmergency' })
+    const StatusDisplayNameMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.statusDisplayName' })
+    const UnitNameMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.unitName' })
+    const UnitNameChangedMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.unitName.change' })
+    const AssigneeMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.assignee' })
+    const ClassifierMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.classifier' })
+    const fields = [
+        ['clientPhone', ClientPhoneMessage],
+        ['details', DetailsMessage],
+        ['clientName', ClientNameMessage],
+        ['isPaid', IsPaidMessage],
+        ['isEmergency', IsEmergencyMessage],
+        ['statusDisplayName', StatusDisplayNameMessage],
+        ['unitName', UnitNameMessage],
+        ['assigneeDisplayName', AssigneeMessage],
+        ['classifierDisplayName', ClassifierMessage],
+    ]
+
+    const BooleanToString = {
+        isPaid: {
+            'true': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isPaid.true' }),
+            'false': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isPaid.false' }),
+        },
+        isEmergency: {
+            'true': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isEmergency.true' }),
+            'false': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isEmergency.false' }),
+        },
+    }
+
+    const formatField = (field, value) => {
+        const formatterFor = {
+            clientPhone: (field, value) => (
+                <PhoneLink value={value}/>
+            ),
+            details: (field, value) => (
+                value.length > 30 ? value.slice(0, 30) + '…' : value
+            ),
+            unitName: (field, value) => (
+                // Formally, unit name (e.g. apartment) was changed, but semantically,
+                // was changed a whole address of the ticket, so, to preserve context,
+                // the change of the unit is displayed as the change of the address
+                UnitNameChangedMessage
+                    .replace('{address}', ticketChange.ticket.property.address)
+                    .replace('{unitName}', value)
+            ),
+        }
+        return has(formatterFor, field)
+            ? formatterFor[field](field, value)
+            : value
+    }
+
+    const format = (field, value) => (
+        typeof value === 'boolean'
+            ? BooleanToString[field][value]
+            : formatField(field, value)
+    )
+
+    /*
+        Interpolates message string with JSX tags.
+        They will be safely mounted in place of `{to}` and `{from}` placeholders
+     */
+    const formatDiffMessage = (field, message, ticketChange) => {
```

Comment:
use useCallback

Reply:
What the use case of `useCallback` in this context?
I does't participates in referential equality checks (does't gets passed to nested component or `useEffects` as a dependency)
it does't memoizes a value
It gets created once for a TicketChange and reused for all changes of fields.
There is no performance workload.

I've measured rendering of 2000 items with and without `useCallback`.
For both cases it takes ~ 10ms.

With `useCallback`:

1: 11.4ms
2: 10.4ms
3: 10.3ms

Without `useCallback`:

1: 12.6ms
2: 10.3ms
3: 10.6ms

You can test by yourself using profiler form React Dev Tools.
[29462c2f](https://github.com/open-condo-software/condo/commit/29462c2f815f39088145b0aea8349c37aadb0cab)

Reply:
Look at memory usage, not at 'ms' render speed, react will reCreate function instance on each rerender.